### PR TITLE
popf: patch src to fix build

### LINKED ISSUE #255: `popf` package - different behaviour of cmake outputs in nix develop and nix build
I'm trying to build humble's `popf` package.

When i run dev environment `nix develop github:lopsided98/nix-ros-overlay?ref=staging#humble.popf` and run in sequence `cmake .` and `make` commands within unpacked release tarball it is successfully configuring and builds.

But when i build package via `nix build github:lopsided98/nix-ros-overlay?ref=staging#humble.popf` configurePhase finished successful, but with CMake warnings:
<details><summary>CMake  warnings in configurePhase</summary>

    CMake Warning: Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DOCDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_INFODIR
    CMAKE_INSTALL_LIBDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR
    CMAKE_POLICY_DEFAULT_CMP0025
    PYTHON_EXECUTABLE
    PYTHON_INCLUDE_DIR
    PYTHON_LIBRARY
</details>

and buildPhase finished with an error:
```
... bash: line 1: /build/popf-release-release-humble-popf-0.0.14-1/src/VALfiles/parsing/fixyywrap: cannot execute: required file not found
``` 

Im not sure, but maybe in can be caused by custom cmake command? (https://github.com/fmrico/popf/blob/humble-devel/CMakeLists.txt#L59):
```
add_custom_command(
       OUTPUT
        ${CMAKE_CURRENT_BINARY_DIR}/pddl+.cpp
       COMMAND
        ${BISON_EXECUTABLE}
       ARGS
        ${CMAKE_CURRENT_SOURCE_DIR}/${parsePDDL_SOURCE_DIR}/pddl+.yacc -o ${CMAKE_CURRENT_BINARY_DIR}/pddl+.cpp
       DEPENDS
        ${CMAKE_CURRENT_SOURCE_DIR}/${parsePDDL_SOURCE_DIR}/pddl+.yacc ${CMAKE_CURRENT_BINARY_DIR}/lex.yy.cc
       COMMAND
        ${CMAKE_CURRENT_SOURCE_DIR}/${parsePDDL_SOURCE_DIR}/fixyywrap ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/${parsePDDL_SOURCE_DIR}/
)
```

### PR DESCRIPTION
**Pull request by movefasta**
Closes #255

### CODE REVIEW

**Review by wentasah — APPROVED** [review]
This looks good. The perl dependency should probably be added upstream to https://github.com/fmrico/popf/blob/master/package.xml, but here we still need to patch the shebang so we cannot avoid the override.

**Comment by movefasta** [comment]
> The perl dependency should probably be added upstream to https://github.com/fmrico/popf/blob/master/package.xml.

There's no perl package in [rosdistro index](https://index.ros.org/search/?term=perl&section=deps). I doubt if we should add it there due a rare using perl in ROS ecosystem. So overlaying in our particular case maybe quite enough.

**Comment by wentasah** [comment]
Right. I think ROS packages can depend on distribution packages, but then the package must be mentioned in https://github.com/ros/rosdistro/blob/master/rosdep/base.yaml. And `perl` is not there, so adding it to `package.xml` makes no sense. Thanks.

File: distros/distro-overlay.nix [hunk 1]
```diff
@@ -171,6 +171,15 @@ let
       nativeBuildInputs = nativeBuildInputs ++ [ self.qt5.wrapQtAppsHook ];
     });
 
+    popf = rosSuper.popf.overrideAttrs ({
+      nativeBuildInputs ? [], ...
```
Comment (lopsided98) [inline distros/distro-overlay.nix:175]:
```suggestion
      nativeBuildInputs ? [], postPatch ? "", ...
```

**Review by lopsided98 — CHANGES_REQUESTED** [review]

**Pull request merged**